include/perfetto/ext/base/scoped_sched_boost.h

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef INCLUDE_PERFETTO_EXT_BASE_SCOPED_SCHED_BOOST_H_
+#define INCLUDE_PERFETTO_EXT_BASE_SCOPED_SCHED_BOOST_H_
+
+#include "perfetto/ext/base/status_or.h"
+#include "perfetto/ext/base/thread_checker.h"
+
+namespace perfetto::base {
+
+struct SchedOsConfig {
+  int policy;
+  int prio;
+};
+
+// kSchedOther: it's the default policy (e.g. CFS on Linux). Range: 0-20.
+//              prio is interpreted as -(nice), i.e. 1 is silhgly higher prio
+//              than the default 0,  20 is the highest priority.
+//              Note that this is the opposite semantic of the cmdline nice, and
+//              is done for consistency with kSchedFifo, so higher
+//              number == higher prio.
+// kSchedFifo: real-time priority. Range: 1-99. 1 is the lowest priority, 99 the
+//             highest.
+struct SchedPolicyAndPrio {
+  enum class Policy {
+    kSchedOther,
+    kSchedFifo
+  };  // Order matters for operator<().
+
+  bool operator<(const SchedPolicyAndPrio& other) const {
+    return std::tie(policy, prio) < std::tie(other.policy, other.prio);
+  }
+
+  bool operator==(const SchedPolicyAndPrio& other) const {
+    return policy == other.policy && prio == other.prio;
+  }
+
+  bool operator!=(const SchedPolicyAndPrio& other) const {
+    return !(*this == other);
+  }
+
+  Policy policy = Policy::kSchedOther;
+  unsigned int prio = 0;
+};
+
+class SchedOsManager {
+ public:
+  static SchedOsManager* GetInstance();
+  // virtual for testing
+  virtual Status SetSchedConfig(const SchedOsConfig& arg);
+  // virtual for testing
+  virtual StatusOr<SchedOsConfig> GetCurrentSchedConfig() const;
+  virtual ~SchedOsManager() = default;
+
+ protected:
+  SchedOsManager() = default;
+
+  SchedOsManager(const SchedOsManager&) = delete;
+  SchedOsManager& operator=(const SchedOsManager&) = delete;
+  SchedOsManager(SchedOsManager&&) = delete;
+  SchedOsManager& operator=(SchedOsManager&&) = delete;
+};
+
+class ThreadMgr {
```

Comment:
same quesiton. why this is exists? why isn't an implementation detail in the anonymous namespace?

Reply:
My bad, the class is now moved to the `scoped_sched_boost.cc` file.